src/util.ts

```diff
@@ -54,3 +54,18 @@ export function toHexString(byteArray: Uint8Array): string {
   });
   return s;
 }
+
+/**
+ * Convert a hex encoded string to a uint8 array
+ *
+ * @param hexString - The string to convert
+ * @returns - The uint8 array, or null in case the input was not a valid
+ * hex-encoded string or empty
+ */
+export function fromHexString(hexString: string): Uint8Array | null {
+  const matches = hexString.match(/.{1,2}/g);
```

Comment:
REQ: The behaviour of this function doesn't match the description. The code happily converts non-hex input into byte arrays, returning bad values. For example passing "za9zz" causes the function to return `[ 0, 9, 0 ]`.

```suggestion
  const matches = hexString.toLowerCase().match(/(0-9a-f){1,2}/g);
```

Unit tests covering this method would be great.